Tests/Reqnroll.VisualStudio.VsxStubs/ProjectSystem/StubIdeScope.cs

```diff
@@ -134,11 +137,15 @@ public Task StartAndWaitAllBackgroundTasks()
         return Task.WhenAll(allTasks);
     }
 
-    private void SetupFireAndForgetOnBackgroundThread()
+    public void SetupFireAndForgetOnBackgroundThread(Action<Func<CancellationToken, Task>, string> callback)
     {
-        Setup(s => s.FireAndForgetOnBackgroundThread(It.IsAny<Func<CancellationToken, Task>>(), It.IsAny<string>()))
-            .Callback((Func<CancellationToken, Task> action, string callerName) =>
-                BackGroundTasks = BackGroundTasks.Add($"{Interlocked.Increment(ref _taskId)}:{callerName}", action));
+        _substitute.When(s => s.FireAndForgetOnBackgroundThread(Arg.Any<Func<CancellationToken, Task>>(), Arg.Any<string>()))
+            .Do(callInfo =>
+            {
```

Comment:
Is this mock setup missing the intended side-effect given in line 141 of the original?
```BackGroundTasks = BackGroundTasks.Add($"{Interlocked.Increment(ref _taskId)}:{callerName}", action));```